intake_esm/source.py

```diff
@@ -0,0 +1,127 @@
+import copy
+
+from intake.source.base import DataSource, Schema
+
+from .merge_util import _aggregate, _path_to_mapper, _to_nested_dict
+
+_DATA_FORMAT_KEY = '_data_format_'
+
+
+class ESMGroupDataSource(DataSource):
+    version = '1.0'
+    container = 'xarray'
+    name = 'esm_group'
+    partition_access = True
+
+    def __init__(
+        self,
+        df,
+        aggregation_dict,
+        path_column,
+        variable_column,
+        data_format=None,
+        format_column=None,
+        cdf_kwargs={'chunks': {}},
+        zarr_kwargs={},
```

Comment:
Do you think we can combine these into a single keyword argument? As far as I understand, they would never be simultaneously utilized.

Reply:
> As far as I understand, they would never be simultaneously utilized.

For catalogs with non-homogeneous assets (netcdf, zarr), you would likely utilize both arguments at the same. So far, I have not seen an example of a non-homogeneous catalog. However, since the esm-collection-spec allows this non-homogeneity, I figured that the easiest thing to do was to separate these two arguments. 

Let me know if you have suggestions on how to combine these two, and still support the non-homogeneous catalogs.

Reply:
Gotcha. I didn't know this was possible. Let's just leave this for now.